testing/asserts.ts

```diff
@@ -202,6 +202,9 @@ export function equal(c: unknown, d: unknown): boolean {
         if (!(a instanceof WeakRef && b instanceof WeakRef)) return false;
         return compare(a.deref(), b.deref());
       }
+      if (a && b && (a.constructor !== b.constructor)) {
```

Comment:
Feels like this should go a lot earlier in the check, as it would short circuit much quicker.

Reply:
I agree. I tried adding it closer to top, but wasn't being too successful (was getting unexpected failures). So I put it at the end as it was the least disruptive place I could find and make the new test pass. This way I could at least get something out and get feedback. Once I get the rest of the tests fixed, I'll try moving it earlier.